### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by Ukaypromise**
### React Capstone Project.

For this project, the following requirements were adhered to:

- [x] Use React documentation.
- [x] Use React components.
- [x] Use React props.
- [x] Use React Router.
- [x] Connect React and Redux.
- [x] Handle events in a React app.
- [x] Write integration tests with a React testing library.
- [x] Use styles in a React app.
- [x] Use React life cycle methods.
- [x] Apply React best practices and language style guides in code.
- [x] Use store, actions and reducers in React

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,47 @@
...
 
 Give a ⭐️ if you like this project!
 
-## Acknowledgments
-
-- Hat tip to anyone whose code was used
-- Inspiration
-- etc
-
 ## 📝 License
```
Comment (roshan-bajgain) [inline README.md]: - [x] I would suggest you please provide appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons](https://creativecommons.org/licenses/) in an Acknowledgements section of your README file

File: README.md [hunk 2]
```diff
@@ -1,75 +1,47 @@
...
-[Live Demo Link](https://livedemo.com)
-
+- Javascript
+- React, Redux
+- Webpack, Babel, React Testing Library
+- Styled Components
 
+## Live Demo
+- Demo: [Not Yet Ready!!!](https://www.loom.com/share/1bb909ddd99a4143b5187c3129c3ab04)
+## Live Video
```
Comment (roshan-bajgain) [inline README.md]: - [x] [Optional] As you said in the comment and I appreciate it so I am going to provide you a link please see and implement it as I am giving it an optional [Link](https://www.youtube.com/watch?v=OPalwvWO63U). please implement it

File: src/features/cryptoSlice.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+export const fetchData = createAsyncThunk(
+  'features/fetchCryptoData',
+  async () => {
+    const res = await fetch(url);
+    const data = await res.json();
+    return data.coins;
+  },
+);
+
```
Comment (roshan-bajgain) [inline src/features/cryptoSlice.js]: - [x] When I try to run the app it gave me an error saying that `Cannot find file: 'CryptoSlice.js' does not match the corresponding name on disk: '.\src\features\cryptoSlice.js'.` So please fix it by renaming the file as `CryptoSlice.js`

File: src/components/CryptoDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                    <h2>Crypto Coin: {coin.name}</h2>
+                    Ranked at: Number{coin.rank}
+                  </div>
+                </div>
+
+                <InfoWrapper>
+                  <Title className="details">
+                    <p>Coin name</p>
+                    <p>{coin.name}</p>
+                  </Title>
```
Comment (roshan-bajgain) [inline src/components/CryptoDetails.jsx]: - [x] For a better user experience please implement the back buttons to navigate to the `Home` page by clicking the "Back" button.
### Reference
hhhhhhhhhhhhhh

File: src/_test_/components.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    expect(UI).toMatchSnapshot();
+  });
+
+  it('Testing for the CryptoDetails component', () => {
+    const UI = render(
+      <Provider store={store}>
+        <Router>
+          <Routes>
+            <Route element={<CryptoDetails />} />
+          </Routes>
```
Comment (roshan-bajgain) [inline src/_test_/components.test.js:29]: - [x] While running the test I got an error as it says ` No routes matched location "/"`. please fix this issue by providing the correct routes
### Error
ssssssssssssssssssssssssssssssssssss

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Ukaypromise, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (Ukaypromise) [reply README.md]: Done!

File: README.md (hunk 2 above)
Comment (Ukaypromise) [reply README.md]: I have used the same methods in the video and i still get deployment failed.

File: src/components/CryptoDetails.jsx (hunk 4 above)
Comment (Ukaypromise) [reply src/components/CryptoDetails.jsx]: Done!

File: README.md (hunk 2 above)
Comment (Ukaypromise) [reply README.md]: zom
How else can I resolve this?

Comment (Ukaypromise) [reply README.md]: @roshan-bajgain

Comment (Ukaypromise) [reply README.md]: Successfully deployed

File: src/features/cryptoSlice.js (hunk 3 above)
Comment (Ukaypromise) [reply src/features/cryptoSlice.js]: Fixed.

File: src/_test_/components.test.js (hunk 5 above)
Comment (Ukaypromise) [reply src/_test_/components.test.js:29]: Fixed

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Ukaypromise 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Amazing job working on both required and optional changes from the previous reviewer.
- Great communication

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**